spec/ics-007-tendermint-client/README.md

```diff
@@ -140,14 +156,14 @@ function checkMisbehaviourAndUpdateState(
     assert(evidence.h1.height === evidence.h2.height)
     // assert that the commitments are different
     assert(evidence.h1.commitmentRoot !== evidence.h2.commitmentRoot)
-    // fetch the previously verified commitment root & validator set hash
+    // fetch the previously verified commitment root & validator set
     consensusState = get("clients/{identifier}/consensusStates/{evidence.fromHeight}")
-    // check that the validator set matches
-    assert(consensusState.validatorSetHash === evidence.fromValidatorSet.hash())
+    // assert that the timestamp is not from more than an unbonding period ago
+    assert(now - consensusState.timestamp < clientState.unbondingPeriod)
     // check if the light client "would have been fooled"
     assert(
-      verify(evidence.fromValidatorSet, evidence.fromHeight, evidence.h1) &&
-      verify(evidence.fromValidatorSet, evidence.fromHeight, evidence.h2)
+      verify(consensusState.validatorSet, evidence.fromHeight, evidence.h1) &&
+      verify(consensusState.validatorSet, evidence.fromHeight, evidence.h2)
       )
     // set the frozen height
```

Comment:
We must also check that frozen height either doesn't already exist or exists for a later height before setting

Reply:
Indeed. Updated!